code/datums/state_machine/state.dm

```diff
@@ -0,0 +1,32 @@
+// An individual state, defined as a `/decl` to save memory.
+// On a directed graph, these would be the nodes themselves, connected to each other by unidirectional arrows.
+/decl/state
+	// Transition decl types, which get turned into refs to those types.
+	// Note that the order DOES matter, as decls earlier in the list have higher priority
+	// if more than one becomes 'open'.
+	var/list/transitions = null
+
+/decl/state/Initialize()
+	. = ..()
+	for(var/i in 1 to LAZYLEN(transitions))
+		var/decl/state_transition/T = GET_DECL(transitions[i])
+		T.from += src
+		transitions[i] = T
+
+// Returns a list of transitions that a FSM could switch to.
+// Note that `holder` is NOT the FSM, but instead the thing the FSM is attached to.
+/decl/state/proc/get_open_transitions(datum/holder)
+	. = list()
```

Comment:
You LAZYLEN() the result of this proc, but it doesn't seem to be a lazylist. Changing `. += T` to `LAZYADD(., T)` and removing the initial list setting would resolve the inconsistency.